modules/encrypt/classes/Kohana/Encrypt/Engine/Openssl.php

```diff
@@ -44,33 +53,43 @@ public function __construct($key_config, $mode = NULL, $cipher = NULL)
 		parent::__construct($key_config, $mode, $cipher);
 
 		$this->_iv_size = openssl_cipher_iv_length($this->_cipher);
-
+		
+		if (! $this->validate_cipher_type())
+		{
+			// No valid encryption cipher is provided!
+			throw new Kohana_Exception('No valid encryption cipher is defined in the encryption configuration. Pick one of AES-***-*** ciphers. ECB type is weak and should not be used.');
+		}
+	}
+	
+	/**
+	 * Check correct cipher validation
+	 * @return bool
+	 */
+	protected function validate_cipher_type():bool
```

Comment:
Specifying the return type of a function is one of the good things of PHP7, but it's not available in other classes in the Kohana core. Maybe we should do this in one of the next milestones and leave it out here. Or should we migrate slowly?

Reply:
Well, for one thing Ko7 should be PHP7 compatible. However all coding should be done according to one standard.